### PR TITLE
Handle circular refs for responses/parameters schemas.

### PR DESCRIPTION
**Pull request by pytlesk4**
- don't prematurely add a parentRef when expanding a response/parameter
- handle transitive refs for responses/parameters

### CODE REVIEW

File: expander.go [hunk 1]
```diff
@@ -1078,7 +1083,7 @@ func expandResponse(response *Response, resolver *schemaLoader, basePath string)
 
 	parentRefs = parentRefs[0:]
```
Comment (fredbi) [inline expander.go]: I never quite understood this line. As far as I remember gometalinter / golangci-lint detect is a useless. Thought?

File: expander.go [hunk 2]
```diff
@@ -770,15 +782,7 @@ func expandSchema(target Schema, parentRefs []string, resolver *schemaLoader, ba
...
-			if transitiveResolver != resolver {
-				debugLog("got a new resolver")
-				if transitiveResolver.options != nil && transitiveResolver.options.RelativeBase != "" {
-					basePath, _ = absPath(transitiveResolver.options.RelativeBase)
-					debugLog("new basePath = %s", basePath)
-				}
-			} else {
-				basePath = normalizedBasePath
-			}
+			basePath = updateBasePath(transitiveResolver, resolver, normalizedBasePath)
```
[the remark this answers is not in the document]
Comment (pytlesk4) [reply expander.go]: correct! Also, we were prematurely pushing into the [parent refs](https://github.com/go-openapi/spec/pull/82/files/2d9ee49688a48852ac27c60d9a9c18d25396cf5f#diff-b01b57fdae3ace684d2bc151252c141eL1159), we only need to do that for the definitions object.

File: expander.go (hunk 1 above)
Comment (pytlesk4) [reply expander.go]: I think it is just a save guard to reset the parent refs array before we expand a schema. I don't know if it is needed, but it is better to be safe I guess.

**Comment by fredbi** [comment]
I am pre-testing all integrated wih go-swagger
validate now is ok.
I am hitting some other problems (that validate does not detect...)
in particular this spec is not valid (hit same issue as #1602)

**Comment by pytlesk4** [comment]
@fredbi after reading that conversation, it doesn't seem like validate was catching that error anyways?

**Comment by fredbi** [comment]
yep it looks okay for that part
other problems with this spec are unrelated

**Pull request merged**